lib/doorkeeper/config.rb

```diff
@@ -56,44 +67,87 @@ def confirm_application_owner
         @config.instance_variable_set('@confirm_application_owner', true)
       end
 
+      # Define default access token scopes for your provider
+      #
+      # @param [Array] scopes (OAuth::Scopes.new) Default set of access
+      # token scopes
       def default_scopes(*scopes)
         @config.instance_variable_set('@default_scopes', OAuth::Scopes.from_array(scopes))
       end
 
+      # Define default access token scopes for your provider
+      #
+      # @param [Array] scopes (OAuth::Scopes.new) Optional set of access
+      # token scopes
       def optional_scopes(*scopes)
         @config.instance_variable_set('@optional_scopes', OAuth::Scopes.from_array(scopes))
       end
 
+      # Change the way client credentials are retrieved from the request object.
+      # By default it retrieves first from the `HTTP_AUTHORIZATION` header, then
+      # falls back to the `:client_id` and `:client_secret` params from the
+      # `params` object.
+      #
+      # @param [Array] methods Define client credentials
```

Comment:
Ok, I like the idea of covering code as much as possible @Morpheu5, so let's try and do that.